db/external_sst_file_test.cc

```diff
@@ -2014,17 +2045,27 @@ TEST_F(ExternalSSTFileTest, FileWithCFInfo) {
 }
 
 /*
- * Test and verify the functionality of ingestion_options.move_files.
+ * Test and verify the functionality of ingestion_options.move_files and
+ * ingestion_options.failed_move_fall_back_to_copy
  */
-TEST_F(ExternalSSTFileTest, LinkExternalSst) {
+TEST_P(ExternSSTFileLinkFailFallbackTest, LinkFailFallBackExternalSst) {
+  const bool fail_link = std::get<0>(GetParam());
+  const bool failed_move_fall_back_to_copy = std::get<1>(GetParam());
+
+  ExternalSSTTestEnv* test_env = new ExternalSSTTestEnv(env_, fail_link);
   Options options = CurrentOptions();
   options.disable_auto_compactions = true;
+  options.env = test_env;
```

Comment:
Seems test_env will be leaked? Can you run the following
```
$COMPILE_WITH_ASAN=1 make -j32 <your test>
$./<your test>
```

Reply:
Thanks. I updated the code to delete the test env.